crates/valence_network/src/lib.rs

```diff
@@ -553,3 +579,39 @@ pub struct PlayerSampleEntry {
     /// The player UUID.
     pub id: Uuid,
 }
+
+async fn do_broadcast_to_lan_loop(shared: SharedNetworkState, motd: String) {
+    loop {
+        let port = shared.0.address.port();
+
+        let Ok(local_addr): Result<SocketAddr, _> = "0.0.0.0:0".parse() else {
+            return;
+        };
+
+        // This weird IP is intentional. It's the multicast address for LAN games.
+        let Ok(server_addr): Result<SocketAddr, _> = "224.0.2.60:4445".parse() else {
+            return;
+        };
+
+        // connect to server_addr on Udp to send a text packet
+        let socket = match std::net::UdpSocket::bind(local_addr) {
+            Ok(s) => s,
+            Err(e) => {
+                println!("couldn't bind socket: {}", e);
+                return;
+            }
+        };
+
+        if socket.connect(server_addr).is_err() {
+            return;
+        }
+
+        let message = format!("[MOTD]{motd}[/MOTD][AD]{port}[/AD]");
+        if socket.send(message.as_bytes()).is_err() {
+            return;
+        }
```

Comment:
Should it though, if there are no clients listening, wouldn't we want to ignore a "failed to connect" ? Although, since it's broadcast, shouldn't the error arm technically be unreachable?